### ISSUE TITLE
Going from SPARQL CONSTRUCT with N-triples to JSON-LD

### ISSUE THREAD

**Issue by fils** [martinmr/gojsonld#1]
I have a simple REST client that makes a CONSTRUCT SPARQL call and returns n-triples.  So the same input test files I see you using.

Is there a way to get from a string into the type Dataset in gojsonld so I can then convert that to JSON-LD via this library?

Thanks

**Comment by martinmr** [comment]
What format is the string in? If it's n-quads, then you can look at parseDataset in rdfDataset.go

Right now, that method is private so you won't be able to use it, but I can change its name to ParseDataset to make it public. If the string is in n-triples, then you would have to convert it into n-quads first by adding a graph element to each RDF statement with the value "@default" so that the statement

s p o .

becomes

s p o @default .

Look at ReadDatasetFromFIle in rdfDatasetUtils.go for an example on how to use parseDataset.

Also, note that the most current version of this repository is at linkeddata/gojsonld

**Comment by martinmr** [comment]
Latest commit with ParseDataset as an exportable/public function is located at https://github.com/linkeddata/gojsonld/commit/a8c991dfb6e6bc4a09e2fe6730dddb04555381dc

**Comment by fils** [comment]
Thanks..  I'll give that a try.  You are right, it's n-triples not n-quads so I will need a default graph or see if I can get Virtuoso to return n-quads for me from a CONSTRUCT call.   

Thanks for the repo note too.   Really looking forward to using this a lot over the next few months on a project.

**Comment by fils** [comment]
Interesting...

My string

```
<http://data.oceandrilling.org/taxa/id/whiteinella/paradubia>   <http://www.w3.org/2004/02/skos/core#related>   <http://foor.org/> .
```

passed through:

```
dataparsed, _ := jsonld.ParseDataset([]byte(sprReturn))
fmt.Println(dataparsed)
```

becomes

```
&{map[@default:[<http://data.oceandrilling.org/taxa/id/whiteinella/paradubia> <http://www.w3.org/2004/02/skos/core#related> <http://foor.org/> .]]}
```

Does your code put in @default for me?

**Comment by martinmr** [comment]
Yeah, it does. I had a look at the code and the parser sets the graph to the default graph (called @default) if it does not find a graph value. This is correct behaviour, so I think the conversion from n-Triples to n-quads might not be needed.

**Comment by fils** [comment]
Great...  that makes my life easier.  I tried:

```
jsonldResults := jsonld.FromRDF(dataparsed, nil)
fmt.Println(jsonldResults)
```

but I get:  panic: runtime error: invalid memory address or nil pointer dereference

I have not looked at the code file you referenced yet though..  so I'm likely doing something stupid.  
I'm really thrilled you made this library by the way...  I was so thrilled to find it.

**Comment by martinmr** [comment]
You need to pass it a reference to a valid Options object. Look at options.go for the set of options. You don't need to set all the options, but the object you pass needs to be valid because it's being dereferenced by FromRDF.

Your code would look like:

``` go
options := &Options{}
jsonldResults := jsonld.FromRDF(dataparsed, options) 
fmt.Println(jsonldResults)
```
